lib/combustion/version_gate.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "bundler"
+require "rubygems"
+
+module Combustion
+  class VersionGate
+    def self.call(name, *constraints)
+      new(name).call(*constraints)
+    end
+
+    def initialize(name)
+      @name = name
+    end
+
+    # Using matches_spec? instead of match? because the former returns true
+    # when the spec has a pre-release version.
+    def call(*constraints)
+      dependency(*constraints).matches_spec?(spec)
+    end
```

Comment:
Should this logic handle if `#spec` is nil, if the thing isn't currently in the bundle at all?  In some cases the combustion seem to deal with the possibility of a thing not being loaded at all (like activerecord could not be loaded even though rails is), although not being loaded might not be the same thing as not being in the gem spec. So maybe that won't come up.  Or for ultimate safety this could be `spec && dependency(...`, so if spec is nil the check is always false?